presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -214,13 +221,12 @@ public void addInput(Page page)
     @Override
     public Page getOutput()
     {
-        if (state != State.FINISHING) {
+        if (state != State.FINISHING || !blocked.isDone()) {
```

Comment:
I think this check is still needed. Otherwise, if this is called while `finishFuture` is not done, we'll block in the call below